windows/src/engine/keyman32/Keyman32.cpp

```diff
@@ -695,9 +695,9 @@ void HandleRefresh(int code, LONG tag)
     // This is sent by Keyman COM API, ApplyToRunningKeymanEngine
     SendDebugMessageFormat(0, sdmGlobal, 0, "#### Refresh Requested ####");
 
-    // We ask any controller window to tell all instances of keyman32/keyman64
-    // that a refresh is coming through
-    Globals::PostControllers(wm_keyman_refresh, KR_PRE_REFRESH, 0);
+    // We ask the master controller window to tell all instances
+    // of keyman32/keyman64 that a refresh is coming through
+    Globals::PostMasterController(wm_keyman_refresh, KR_PRE_REFRESH, 0);
```

Comment:
There is no need to ask any controller, just ask the master controller, and this reduces duplicate refresh requests yet further.